src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/strategy/AbstractStrategy.java

```diff
@@ -0,0 +1,160 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.strategy;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdAuthenticator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdAuthenticatorFactory;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Classification;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+/**
+ * Implementation of strategy.
+ * Any strategy is invoking steps as follows:
+ * <ul>
+ *     <li>jurisdiction authentication</li>
+ *     <li>start the event</li>
+ *     <li>create data content accordingly</li>
+ *     <li>submit the event</li>
+ * </ul>
+ * Any strategy will have to be implemented as an example:
+ * <pre>{@code
+ * @Component("strategy-name")
+ * class StrategyName extends AbstractStrategy {
+ *
+ *     StrategyName() {
+ *         // any extra autowiring needed
+ *     }
+ *
+ *     @Override
+ *     public boolean isStrategyEligible(Classification classification, boolean caseExists) {
+ *         return false; // by default it returns false so override in case strategy needs any rules to apply
+ *     }
+ *
+ *     @Override
+ *     Object mapEnvelopeToCaseDataObject(Envelope envelope) {
+ *         return null; // implement this
+ *     }
+ *
+ *     @Override
+ *     String getEventTypeId() {
+ *         return ""
+ *     }
+ *
+ *     @Override
+ *     String getEventSummary() {
+ *         return ""
+ *     }
+ * }}</pre>
+ * <p/>
+ * No need to make any public access as everything will be injected in {@link StrategyContainer}:
+ * <pre>{@code
+ * @Resource(name = "strategy-name")
+ * private Strategy someNamedStrategy;}</pre>
+ * <p/>
+ * Then include each strategy in {@link StrategyContainer#setUpStrategies()}
+ */
+abstract class AbstractStrategy implements Strategy {
+
+    private static final String CASE_TYPE_ID = "Bulk_Scanned";
+
+    private static final Logger log = LoggerFactory.getLogger(AbstractStrategy.class);
+
+    @Autowired
+    private CoreCaseDataApi ccdApi;
+
+    @Autowired
+    private CcdAuthenticatorFactory authenticatorFactory;
+
+    @Override
+    public boolean isStrategyEligible(Classification classification, boolean caseExists) {
+        return false;
+    }
+
+    @Override
+    public void execute(Envelope envelope) {
```

Comment:
This implementation is quite strict. Is it always going to work like this?

Reply:
This is the idea. It will be always the same. If anything differs - can be easily overridden by lifting `private` access where needed.